app/src/main/kotlin/compiler/intermediate/generators/DefaultFunctionDetailsGenerator.kt

```diff
@@ -35,6 +35,7 @@ data class DefaultFunctionDetailsGenerator(
     private val variablesRegisters: MutableMap<NamedNode, Register> = referenceHashMapOf()
     private var variablesTotalOffset: ULong = 0u
     private val previousDisplayEntryRegister = Register()
+    private val calleeSavedBackupRegisters = calleeSavedRegistersWithoutRSPAndRBP.map { Register() }
```

Comment:
Since we are at it, why not backup RBP in this way as well?

Reply:
Old RBP must be pushed to the stack right below return address (according to the calling convention).